### DISCUSSION TITLE
Redesigning VNI allocation

### DISCUSSION THREAD

**Discussion by ipspace** [ipspace/netlab#509] [Ideas]
With the benefits of perfect hindsight: we never should have allocated VNI values in the VLAN module. It seemed like a good idea at that time, but turned out to be less-than-ideal now that I'm considering EPVN-over-MPLS.

So here's the new idea (that I would introduce in 1.4 -- see #510):

* VLAN module allocates **vlan.id** values
* VXLAN module uses the list of VXLAN-enabled VLAN modules to allocate **vlan.vni** values
* The default list of VXLAN modules (when missing) is "all VLANs", replicating the existing behavior with **vlan.auto_vni** set to True
* Using an explicit list of VXLAN-enabled VLANs is equivalent to **vlan.auto_vni** set to False
* Static **vlan.vni** values (integer or boolean) will still be supported
* At the end of the transformation, the node **vxlan.vlans** list will reflect the VXLAN-enabled VLANs (those with valid VNI values)

Any objections or suggestions @ssasso @jbemmel?

**Comment by jbemmel** [comment]
See https://github.com/ipspace/netlab/issues/317 - I've always felt that the ```vxlan``` module would be a better place for VNIs

**Reply by ipspace** [reply]
And you were right ;)

What about the **auto_vni**. Are you OK with dropping it and replacing it with an explicit list of VLANs?
